python/ql/src/semmle/python/dataflow/new/internal/DataFlowPrivate.qll

```diff
@@ -544,6 +546,27 @@ class DataFlowCallableValue extends DataFlowCallable, TCallableValue {
   override CallableValue getCallableValue() { result = callable }
 }
 
+/** A class representing a callable lambda. */
+class DataFlowLambda extends DataFlowCallable, TLambda {
+  Function lambda;
+
+  DataFlowLambda() { this = TLambda(lambda) }
+
+  override string toString() { result = lambda.toString() }
+
+  override CallNode getACall() { result = getCallableValue().getACall() }
+
+  override Scope getScope() { result = lambda.getEvaluatingScope() }
+
+  override NameNode getParameter(int n) { result = getParameter(getCallableValue(), n) }
+
+  override string getName() { result = "Lambda callable" }
+
+  override CallableValue getCallableValue() {
+    result.(FunctionValue).getOrigin().getNode() = lambda.getDefinition()
+  }
```

Comment:
```suggestion
  override FunctionValue getCallableValue() {
    result.getOrigin().getNode() = lambda.getDefinition()
  }
```
Looking at this a second time, I _think_ this should be allowed by the compiler.